pkg/azclient/cache/go.mod

```diff
@@ -1,6 +1,8 @@
 module sigs.k8s.io/cloud-provider-azure/pkg/azclient/cache
 
-go 1.23.2
+go 1.23.0
+
+toolchain go1.24.0
```

Comment:
why GO 1.23 using 1.24 toolchain here?

Reply:
to mitigate go cve. please see @kaovilai 's comment in issue #8535

Reply:
will take a look thx!

Reply:
Just for context.. `1.24` or `1.24.0` is also fine. What I want to avoid is use of `go <go-directive>` bump for CVEs.

CVE should use separate toolchain line to prevent dependents compatibility breakages., keeping go directive with patch .0 or not at all.. ie. `1.24` or `1.24.0`.

Toolchain only affect the curent repo and leave it up to depedents if they want to match it.